### PR TITLE
Bluetooth: Controller: Consolidated set of pending changes

### PR DESCRIPTION
**Pull request by cvinayak**
Set of commits that has fixed assorted test failures in conformance testing.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ull_adv_sync.c [hunk 1]
```diff
@@ -231,25 +263,176 @@ uint8_t ll_adv_sync_ad_data_set(uint8_t handle, uint8_t op, uint8_t len,
...
+		/* Add AD Data and remove any prior presence of Aux Ptr */
+		err = ull_adv_sync_pdu_set_clear(lll_sync, pdu_prev, pdu,
+						 ULL_ADV_PDU_HDR_FIELD_AD_DATA,
+						 ULL_ADV_PDU_HDR_FIELD_AUX_PTR,
+						 hdr_data);
+#if defined(CONFIG_BT_CTLR_ADV_SYNC_PDU_LINK)
+		if (!err) {
+			/* Fragment into chain PDU if len > 191 bytes */
+			if (len > (PDU_AC_PAYLOAD_SIZE_MAX -
+				   PDU_AC_EXT_PAYLOAD_OVERHEAD)) {
```
Comment (ppryga) [inline subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:306]: This way to calculate max PDU size for a single extended advertising PDU is used at leas in two places `PDU_AC_PAYLOAD_SIZE_MAX - PDU_AC_EXT_PAYLOAD_OVERHEAD1`. Maybe use of a single named macro that covers the subtraction? I would improve readability.

File: subsys/bluetooth/controller/ll_sw/ull_adv_sync.c [hunk 2]
```diff
@@ -277,6 +460,168 @@ uint8_t ll_adv_sync_ad_data_set(uint8_t handle, uint8_t op, uint8_t len,
...
+		pdu_chain->len = 0U;
+
+		com_hdr_chain = &pdu_chain->adv_ext_ind;
+		hdr_chain = (void *)&com_hdr_chain->ext_hdr_adv_data[0];
+		dptr_chain = (void *)hdr_chain;
+
+		/* Flags */
+		*dptr_chain = 0U;
+
+		/* TODO: should CTEInfo be added to chain PDU due to AD data
```
Comment (ppryga) [inline subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:508]: CTE count is given by `HCI_LE_Set_Connectionless_CTE_Transmit_Parameters`, hence it is not altered due to change on PDUs count in periodic advertising chain.

File: subsys/bluetooth/controller/ll_sw/ull_adv_sync.c [hunk 3]
```diff
@@ -277,6 +460,168 @@ uint8_t ll_adv_sync_ad_data_set(uint8_t handle, uint8_t op, uint8_t len,
...
+			dptr_chain++;
+		}
+
+		/* Start adding fields corresponding to flags here, if any */
+
+		/* No AdvA flag */
+		/* No TgtA flag */
+
+		/* CTEInfo flag */
+		if (hdr_chain->cte_info) {
```
Comment (ppryga) [inline subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:531]: How new PDU may have any of these flags already filled? It is allocated in line 490.

File: subsys/bluetooth/controller/ll_sw/ull_adv_sync.c [hunk 4]
```diff
@@ -277,6 +460,168 @@ uint8_t ll_adv_sync_ad_data_set(uint8_t handle, uint8_t op, uint8_t len,
...
+			 * new chain PDU
+			 */
+			(void)memcpy(&ad_overflow,
+				     &val_ptr[ULL_ADV_HDR_DATA_DATA_PTR_OFFSET],
+				     sizeof(ad_overflow));
+			ad_overflow += *val_ptr;
+			(void)memcpy(dptr_chain, ad_overflow, ad_len_overflow);
+			dptr_chain += ad_len_overflow;
+
+			/* Reduce the AD data in the previous PDU */
```
Comment (ppryga) [inline subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:569]: "in the previous PDU"? Comment in lines 559-561 says about reduction of a AD data in current PDU.

File: subsys/bluetooth/controller/ll_sw/ull_adv_sync.c [hunk 5]
```diff
@@ -1069,18 +1429,28 @@ uint8_t ull_adv_sync_pdu_set_clear(struct lll_adv_sync *lll_sync,
...
-		if (ter_hdr.adi) {
-			ter_dptr += sizeof(struct pdu_adv_adi);
-		}
-		if (ter_hdr_prev.adi) {
-			ter_dptr_prev += sizeof(struct pdu_adv_adi);
-		}
+	/* ADI */
+	if (hdr_add_fields & ULL_ADV_PDU_HDR_FIELD_ADI) {
+		ter_hdr.adi = 1U;
+		/* return the size of ADI structure */
```
Comment (ppryga) [inline subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:1457]: Missing information in a function description.

File: subsys/bluetooth/controller/ll_sw/ull_adv_sync.c [hunk 6]
```diff
@@ -1069,18 +1429,28 @@ uint8_t ull_adv_sync_pdu_set_clear(struct lll_adv_sync *lll_sync,
...
+		*(uint8_t *)hdr_data = sizeof(struct pdu_adv_adi);
+		hdr_data = (uint8_t *)hdr_data + sizeof(uint8_t);
+		/* pick the reference to ADI param */
+		(void)memcpy(&adi, hdr_data, sizeof(struct pdu_adv_adi *));
+		/* return the pointer to ADI struct inside the PDU */
```
Comment (ppryga) [inline subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:1462]: Missing information in a function description.

**Review by ppryga-nordic — CHANGES_REQUESTED** [review]
I've done a very brief review of all commits except this one: b905598de4b1e0ec89278e78ff444b6775493f8f.

File: subsys/bluetooth/controller/ll_sw/ull_adv_aux.c [hunk 7]
```diff
@@ -1332,6 +1332,134 @@ void ull_adv_aux_done(struct node_rx_event_done *done)
...
+
+		/* Get the next chain PDU */
+		pdu_chain_prev = lll_adv_pdu_linked_next_get(pdu_prev);
+		if (!pdu_chain_prev) {
+			break;
+		}
+
+		/* Fill the aux offset in the parent PDU */
+		offs_us = PDU_AC_US(pdu->len, phy_s, phy_flags) + mafs_us;
+		ull_adv_aux_ptr_fill(aux_ptr, offs_us, phy_s);
```
[the remark this answers is not in the document]
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_adv_aux.c]: added function description

**Review by kruithofa — APPROVED** [review]

**Review by ppryga-nordic — APPROVED** [review]

**Pull request merged**